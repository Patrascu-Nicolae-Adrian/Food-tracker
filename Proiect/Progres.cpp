#include "Progres.h"
#include <string>
#include <iostream>
using namespace std;

void Progres::setConsume(double consum) { consume = consum; }
double Progres::getConsume() { return consume; }
void Progres::setProtein(double pro) { protein = pro; }
double Progres::getProtein() { return protein; }
void Progres::setCarbohydrate(double carbo) { carbohydrate = carbo; }
double Progres::getCarbohydrate() { return carbohydrate; }
void Progres::setFat(double f) { fat = f; }
double Progres::getFat() { return fat; }
Progres::Progres(double consume, double protein, double carbohydrate, double fat) {
	this->consume = consume;
	this->protein = protein;
	this->carbohydrate = carbohydrate;
	this->fat = fat;
}
int transform1(char*& token, char*& next_token)
{

	const char progres_delim[4] = { ",><" };
	double a = atof(token);
	token = strtok_s(NULL, progres_delim, &next_token);
	return a;
}
Progres::Progres(char line[]) {
	char* next_token = NULL;
	const char progres_delim[4] = { ",><" };
	char* token = strtok_s(line, progres_delim, &next_token);
	consume = transform1(token, next_token);
	protein = transform1(token, next_token);
	carbohydrate = transform1(token, next_token);
	fat = transform1(token, next_token);
}
Progres::Progres() {
	consume = 0;
	protein = 0;
	carbohydrate = 0;
	fat = 0;
}

std::ostream& operator<<(std::ostream& out, Progres p) {
	out << "Consume: " << p.getConsume() << endl;
	out << "Protein: " << p.getProtein() << endl;
	out << "Carbohydrate: " << p.getCarbohydrate() << endl;
	out << "Fat: " << p.getFat() << endl;
	return out;
}
Progres Progres::operator+(Progres p)
{
	Progres k;
	k.setConsume(getConsume() + p.getConsume());
	k.setProtein(getProtein() + p.getProtein());
	k.setCarbohydrate(getCarbohydrate() + p.getCarbohydrate());
	k.setFat(getFat() + p.getFat());
	return k;
}
Progres Progres::operator%(int p)
{
	Progres k;
	k.setConsume(getConsume() * p / 100);
	k.setProtein(getProtein() * p / 100);
	k.setCarbohydrate(getCarbohydrate() * p / 100);
	k.setFat(getFat() * p / 100);
	return k;
}
