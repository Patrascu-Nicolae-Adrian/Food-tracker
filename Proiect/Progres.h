#pragma once
#include <iostream>
class Progres
{
public:
	void setConsume(double );
	double getConsume();
	void setProtein(double );
	double getProtein();
	void setCarbohydrate(double );
	double getCarbohydrate();
	void setFat(double );
	double getFat();
	Progres(char s[]);
	Progres();
	Progres(double, double, double, double);
	friend std::ostream& operator<<(std::ostream&, Progres);
	Progres operator+(Progres);
	Progres operator%(int);
private:
	double consume;
	double protein;
	double carbohydrate;
	double fat;
};

