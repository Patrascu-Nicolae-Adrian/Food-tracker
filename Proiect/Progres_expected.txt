12.02.2020
Mic_dejun
400,50,20,30,30,20,1,3
Pranz
700,40,30,60,60,30,3,0
Cina
500,30,40,50,40,10,2,2
Gustari
200,20,10,20,20,5,1,20
Total
1800,140,100,160,150,65,7,25
13.02.2020
Mic_dejun
700,50,10,30,20,10,2,2
Pranz
400,40,30,60,60,40,3,0
Cina
500,30,50,50,50,10,1,2
Gustari
300,20,10,20,20,5,1,20
Total
1800,140,100,160,150,65,7,25
14.02.2020
Mic_dejun
200,50,20,50,30,20,1,3
Pranz
900,40,40,60,60,40,3,0
Cina
300,20,30,30,40,20,2,2
Gustari
400,30,10,20,20,5,3,20
Total
1800,140,100,160,150,85,9,25